airmozilla/manage/related.py

```diff
@@ -0,0 +1,44 @@
+import pyelasticsearch
+import datetime
+
+from django.conf import settings
+from django.utils import timezone
+
+from airmozilla.main.models import Event
+
+
+def index(all=False, flush_first=False, since=datetime.timedelta(minutes=10)):
+    es = pyelasticsearch.ElasticSearch(settings.RELATED_CONTENT_URL)
+
+    if flush_first:
+        try:
+            es.flush(
+                settings.ELASTICSEARCH_PREFIX + settings.ELASTICSEARCH_INDEX
+            )
+        except pyelasticsearch.exceptions.ElasticHttpNotFoundError:
+            # if the index isn't there we can't flush it
+            pass
+
+    if all:
+        events = Event.objects.scheduled_or_processing()
+    else:
+        now = timezone.now()
+        events = Event.objects.scheduled_or_processing() \
+            .filter(modified__gte=now-since)
+
+    for event in events:
+        # should do bulk ops
+        es.index(
+            settings.ELASTICSEARCH_PREFIX + settings.ELASTICSEARCH_INDEX,
+            'event',
+            {
+                'title': event.title,
+                'privacy': event.privacy,
+                'tags': [x.name for x in event.tags.all()],
+                'channels': [x.name for x in event.channels.all()],
+            },
+            id=event.id,
+
+        )
```

Comment:
So I think it's time to start thinking about doing a bulk index instead. It's going to make little difference if there's only 1-5 events that needs to be indexed but it's going to make a HUGE difference when we ALL. 

What we need to do is make two bulk updates. 
Something like this:
```python
if all:
    events = Event.objects.scheduled_or_processing()
    removed = Event.objects.filter(status=Event.STATUS_REMOVED)
else:
    now = timezone.now()
    events = Event.objects.scheduled_or_processing() \
            .filter(modified__gte=now-since)
    removed = Event.objects.filter(status=Event.STATUS_REMOVED)\
             .filter(modified__gte=now-since)

events_as_list = [
    {
                'id': event.id,
                'title': event.title,
                'privacy': event.privacy,
                'tags': [x.name for x in event.tags.all()],
                'channels': [x.name for x in event.channels.all()],
            }
         for event in events
]
es.bulk(
    (es.index_op(doc, id=doc.pop('id')) for doc in events_as_list),
    index=settings.ELASTICSEARCH_PREFIX + settings.ELASTICSEARCH_INDEX,
    doc_type='event'
)
for event in removed:
    es.delete_op(id=event.id)
```